src/module/iotjs_module_buffer.c

```diff
@@ -0,0 +1,374 @@
+/* Copyright 2015-2016 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "iotjs_def.h"
+#include "iotjs_module_buffer.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+
+iotjs_bufferwrap_t* iotjs_bufferwrap_create(const iotjs_jval_t* jbuiltin,
+                                            size_t length) {
+  iotjs_bufferwrap_t* bufferwrap = IOTJS_ALLOC(iotjs_bufferwrap_t);
+  IOTJS_VALIDATED_STRUCT_CONSTRUCTOR(iotjs_bufferwrap_t, bufferwrap);
+
+  iotjs_jobjectwrap_initialize(&_this->jobjectwrap,
+                               jbuiltin,
+                               (uintptr_t)bufferwrap,
+                               (JFreeHandlerType)iotjs_bufferwrap_destroy);
+  if (length > 0) {
+    _this->buffer = iotjs_buffer_allocate(length);
+    IOTJS_ASSERT(_this->buffer != NULL);
+  } else {
+    _this->buffer = NULL;
+  }
+  _this->length = length;
```

Comment:
IMHO, if length is lees than 0, ``_this->length`` will get a negative value. Don't we need to check it?